cedar-policy-core/src/est/expr.rs

```diff
@@ -895,13 +934,27 @@ impl TryFrom<&Node<Option<cst::Relation>>> for Expr {
             }
             cst::Relation::Like { target, pattern } => {
                 let target_expr = target.try_into()?;
-                let pat_expr: Expr = pattern.try_into()?;
-                let pat_str = pat_expr.into_string_literal().map_err(|e| {
-                    pattern.to_ast_err(ToASTErrorKind::InvalidPattern(
-                        serde_json::to_string(&e).unwrap_or_else(|_| "<malformed est>".to_string()),
-                    ))
-                })?;
-                Ok(Expr::like(target_expr, pat_str))
+                let mut errs = ParseErrors::new();
+                match pattern
+                    .to_expr_or_special(&mut errs)
+                    .map(|expr| expr.into_pattern(&mut errs))
+                {
+                    Some(Some(pat)) => Ok(Expr::like(
+                        target_expr,
+                        pat.into_iter().map(PatternElem::from),
+                    )),
+                    _ => {
+                        if errs.is_empty() {
+                            Err(pattern
+                                .to_ast_err(ToASTErrorKind::InvalidPattern(
```

Comment:
This case is kind of odd. I _think_ it should only ever happen if `pattern.ok_or_missing()?` returns an `Err` result, so the `InvalidPattern` error shouldn't be reachable

Reply:
Yes, it makes sense. Revised.